### ISSUE TITLE
Exclusive windows keep raising themselves

### ISSUE THREAD

**Issue by Meriipu** [godotengine/godot#40024]
**Godot version:**
8a484756de0d66aa91da7975672234d58ca31d7d

**OS/device including version:**
Linux

**Issue description:**
If I move a non-exclusive window (e.g. the main window) to the front after a dialogue has opened, it almost immediately gets pushed back again.

**Steps to reproduce:**
* Start godot (project manager)
* click scan (a window to pick which directory to scan in should open)
* move the parent project manager window back to the top
* on the next update frame, the scan-window autoraises itself

Very few or no applicaitons I use have this behaviour. My browsers (firefox and opera) do not have this behaviour.
The browsers _do_ stop input from reaching the parent window (as godot too stops), but I can still raise the parent window to examine its current state. 

in short please stop automatically raising windows which have been lowered

**Comment by Anutrix** [comment]
Firefox doesn't allow the user to raise the parent window to examine its current state unless the popup is done(closed or action completed). I tried with "File Upload" and "Close all tabs".
I tested this on Opera GX too and its the same.
I am on Windows.

**Comment by Meriipu** [comment]
> Firefox doesn't allow the user to raise the parent window to examine its current state unless the popup is done(closed or action completed). I tried with "File Upload" and "Close all tabs" popup.
> I tested this on Opera GX too and its the same.
> I am on Windows.

Might it be the Windows window manager that does not allow it?
Does the Windows window manager allow you to raise the parent window?
I use ctwm (a twm derivative) on Linux, and it lets me raise parent windows just fine.

**Comment by Anutrix** [comment]
> 
> 
> > Firefox doesn't allow the user to raise the parent window to examine its current state unless the popup is done(closed or action completed). I tried with "File Upload" and "Close all tabs" popup.
> > I tested this on Opera GX too and its the same.
> > I am on Windows.
> 
> Might it be the Windows window manager that does not allow it?
> Does the Windows window manager allow you to raise the parent window of Godot?
> I use ctwm (a twm derivative) on Linux, and it lets me raise parent windows just fine.
> 
> edit: forgot to specify Godot.

Godot behaves the same way as Firefox and OperaGX on Windows i.e, it doesn't allow the parent window to be raised.

**Comment by Meriipu** [comment]
> Godot behaves the same way as Firefox and OperaGX on Windows i.e, it doesn't allow the parent window to be raised.

it is probably your window manager that does not allow it, then?
I do not see why godot should autoraise its windows to try to step on the toes of window managers that do allow raising the parent window (even if the parent window will not receive input). 

Either disallow it completely if possible, or never autoraise.

**Comment by Meriipu** [comment]
Appears to be fixed in 4.0
